api/src/main/java/jakarta/data/repository/CrudRepository.java

```diff
@@ -38,35 +38,48 @@ public interface CrudRepository<T, K> extends BasicRepository<T, K> {
+     * @param <S> Type of the entity to insert.
+     * @return the inserted entity, which may or may not be a different instance depending on the Jakarta Data provider.
+     * @throws EntityExistsException if the entity is already present in the database (in ACID-supported databases).
      * @throws NullPointerException if the entity is null.
-     * @throws UnsupportedOperationException for Key-Value and Wide-Column databases
-     *         that use an append model to write data.
      */
-    void insert(T entity);
+    <S extends T> S insert(S entity);
 
     /**
-     * <p>Inserts multiple entities into the database. If an entity of this type with the same
-     * unique identifier as any of the given entities already exists in the database,
-     * then this method raises {@link EntityExistsException}.</p>
+     * <p>Inserts multiple entities into the database. If any entity of this type with the same
+     * unique identifier as any of the given entities already exists in the database and the database
+     * supports ACID transactions, then this method raises {@link EntityExistsException}.
+     * In databases that follow the BASE model or use an append model to write data, this exception
+     * may not be thrown.</p>
+     *
+     * <p>The entities within the returned iterable may be the same instances as those supplied
+     * as parameters, especially in non-Java record classes. However, for Jakarta Data providers
+     * that support Java records, different instances may be returned.</p>
      *
      * @param entities entities to insert.
-     * @throws EntityExistsException if any of the entities are already present in the database.
-     * @throws NullPointerException if either the iterable is null or any element is null.
-     * @throws UnsupportedOperationException for Key-Value and Wide-Column databases
-     *         that use an append model to write data.
+     * @param <S> Type of the entities to insert.
+     * @return an iterable containing the inserted entities, which may or may not be different instances depending on the Jakarta Data provider.
+     * @throws EntityExistsException if any of the entities are already present in the database (in ACID-supported databases).
+     * @throws NullPointerException if the iterable is null or any element is null.
      */
-    void insertAll(Iterable<T> entities);
+    <S extends T> Iterable<S> insertAll(Iterable<S> entities);
 
     /**
      * <p>Modifies an entity that already exists in the database.</p>
      *
      * <p>For an update to be made, a matching entity with the same unique identifier
-     * must be present in the database.</p>
+     * must be present in the database. In databases that use an append model to write data or
+     * follow the BASE model, this method will work similarly to the {@link #insert} method,
+     * especially if the database does not support ACID transactions.</p>
```

Comment:
A specification cannot have ambiguous language like "will work similarly to the {@link #insert} method, especially".  This doesn't have any enforceable meaning whatsoever because "similarly" is a matter of interpretation.  A specification needs to be precise about stating requirements. Otherwise each vendor will implement its own interpretation, and customers will have no guarantees against which to code and so be unable to use the API.  Also, language like "especially" should never be used in a spec requirement because it suggests that it is actually a non-requirement.

Reply:
Thank you, I will update this point.

Reply:
Could you check again?

Reply:
Switching from "especially" to "particularly" and from "will work" to "behaves" doesn't change anything about clearly stating requirements in the spec.  This still lacks explicit requirements and leaves the reader guessing what it means that works/behaves similarly to insert.   My prior comment still applies.

Reply:
Do you have any suggestions?
Otherwise, it is time to discuss a SQLRepository.

Reply:
I did a new update, the challenge here:

Appendedable database and BASE databases, both have update methods, but while Appendable works as an alias to both update and insert with BASE the update will be eventual consistency,